src/energyplus/ForwardTranslator/ForwardTranslateZoneHVACEvaporativeCoolerUnit.cpp

```diff
@@ -0,0 +1,254 @@
+        } else {
+          outletNodeName = coolerOutletNodeName.get();
+        }
+      } else {
+        if (secondEvaporativeCooler_) {
+          inletNodeName = baseName + " Second Evaporative Cooler - Fan Node";
+        } else {
+          inletNodeName = baseName + " First Evaporative Cooler - Fan Node";
+        }
+        outletNodeName = coolerOutletNodeName.get();
+      }
+
+      if (fan_->iddObject().type() == IddObjectType::Fan_ConstantVolume) {
+        fan_->setString(Fan_ConstantVolumeFields::AirInletNodeName, inletNodeName);
+        fan_->setString(Fan_ConstantVolumeFields::AirOutletNodeName, outletNodeName);
+      } else if (fan_->iddObject().type() == IddObjectType::Fan_VariableVolume) {
+        fan_->setString(Fan_VariableVolumeFields::AirInletNodeName, inletNodeName);
+        fan_->setString(Fan_VariableVolumeFields::AirOutletNodeName, outletNodeName);
+      } else if (fan_->iddObject().type() == IddObjectType::Fan_OnOff) {
+        fan_->setString(Fan_OnOffFields::AirInletNodeName, inletNodeName);
+        fan_->setString(Fan_OnOffFields::AirOutletNodeName, outletNodeName);
+      } else if (fan_->iddObject().type() == IddObjectType::Fan_SystemModel) {
+        fan_->setString(Fan_SystemModelFields::AirInletNodeName, inletNodeName);
+        fan_->setString(Fan_SystemModelFields::AirOutletNodeName, outletNodeName);
+      } else if (fan_->iddObject().type() == IddObjectType::Fan_ComponentModel) {
+        fan_->setString(Fan_ComponentModelFields::AirInletNodeName, inletNodeName);
+        fan_->setString(Fan_ComponentModelFields::AirOutletNodeName, outletNodeName);
+      }
+    }
+
+    if (firstEvaporativeCooler_) {
+      std::string outletNodeName;
+      std::string inletNodeName;
+      if (istringEqual(fanPlacement, "BlowThrough") && fan_) {
+        inletNodeName = baseName + " Fan - First Evaporative Cooler Node";
+      } else {
+        inletNodeName = outdoorAirInletNodeName;
+      }
+
+      if (secondEvaporativeCooler_) {
+        outletNodeName = baseName + " First Evaporative Cooler - Second Evaporative Cooler Node";
+      } else if (istringEqual(fanPlacement, "BlowThrough")) {
+        outletNodeName = coolerOutletNodeName.get();
+      } else {
+        outletNodeName = baseName + " First Evaporative Cooler - Fan Node";
+      }
+
+      if (firstEvaporativeCooler_->iddObject().type() == IddObjectType::EvaporativeCooler_Direct_ResearchSpecial) {
+        firstEvaporativeCooler_->setString(EvaporativeCooler_Direct_ResearchSpecialFields::AirInletNodeName, inletNodeName);
+        firstEvaporativeCooler_->setString(EvaporativeCooler_Direct_ResearchSpecialFields::AirOutletNodeName, outletNodeName);
+        firstEvaporativeCooler_->setString(EvaporativeCooler_Direct_ResearchSpecialFields::SensorNodeName, outletNodeName);
+      } else if (firstEvaporativeCooler.iddObject().type() == IddObjectType::EvaporativeCooler_Indirect_ResearchSpecial) {
```

Comment:
UH OH. Checking the type of the Modelobject!!! Which is "OS:xxxx" and WILL NOT MATCH.

I revamped the OS-resources test to flip the order of the Indirect and Direct evap cooler so it's similar to https://github.com/NREL/EnergyPlus/blob/31e3c33467c5873371bf48b12a7318215971c315/testfiles/StripMallZoneEvapCoolerAutosized.idf#L4767-L4784

And I was getting empty nodes, and I couldn't figure out why. I just noticed this only after completely rewrittign the FT routine...